src/sdk/base/publication.js

```diff
@@ -33,7 +40,15 @@ export class AudioPublicationSettings {
  */
 export class VideoPublicationSettings {
   // eslint-disable-next-line require-jsdoc
-  constructor(codec, resolution, frameRate, bitrate, keyFrameInterval, rid) {
+  constructor(id, codec, resolution, frameRate,
+      bitrate, keyFrameInterval, rid) {
+    /**
+     * @member {?Owt.Base.VideoCodecParameters} id
+     * @instance
+     * @classDesc Identifer for this set of publication settings.
+     * @memberof Owt.Base.VideoPublicationSettings
+     */
+    this.id=id,
```

Comment:
Similar to audio, I don't think ID is required here. You may refer to the WebRTC spec for the concept of constraints, settings and capabilities.

Reply:
If a publication has 2 video tracks without rid but mid, current members cannot distinguish them in an array of videoPublicationSettings.